### PR TITLE
fix ratio

### PR DESCRIPTION
**Pull request by hhaAndroid**
the loss calculation will convert logits to float32, so for alignment, we also need to convert it to float32 here to prevent the ratio from being 1 during rl training

### CODE REVIEW

File: xtuner/v1/module/lm_head/lm_head.py [hunk 1]
```diff
@@ -33,7 +33,9 @@ def forward(  # type: ignore[override]
             b = self.bias
         if loss_ctx is None:
             logits = F.linear(hidden_states, w, b)
-            return None, logits
+            # Note: the loss calculation will convert logits to float32, so for alignment,
+            # we also need to convert it to float32 here to prevent the ratio from being 1 during rl training
```
Comment (RangiLyu) [inline xtuner/v1/module/lm_head/lm_head.py:37]:
```suggestion
            # we also need to convert it to float32 to prevent the ratio from not being equal to 1 during on-policy rl training
```

**Comment by pppppM** [comment]
Due to the lack of use of chunk forward when computing old_logprobs, it may cause OOM (Out of Memory).
From the perspective of inference, the logits returned need to be bf16. 
It is best to convert to float32 near the computation of logprobs.
